src/cljc/witan/schema.cljc

```diff
@@ -19,3 +20,101 @@
   "validation for /login"
   {(s/required-key :username) (s/both (length-greater 5) (is-an-email))
    (s/required-key :password) (length-greater 5)})
+
+(def IdType
+  s/Str)
+
+(def DateTimeType
+  sc/ISO-Date-Time)
+
+(def User
+  "User"
+  {(s/required-key :id)   IdType
+   (s/required-key :name) s/Str}) ;; TODO add groups
+
+(def ModelPropertyType
+  "Types that the model properties can be"
+  (s/enum
+   :text
+   :number
+   :dropdown))
+
+(def ModelProperty
+  "Properties that a model can expose"
+  {(s/required-key :id)      IdType
+   (s/required-key :name)    s/Str
+   (s/required-key :type)    ModelPropertyType
+   (s/optional-key :context) s/Any}) ;; varies depending on the type
+
+(def Model
+  "Models are the center-piece of a Forecast"
+  {(s/required-key :series)     IdType
+   (s/required-key :name)       s/Str
+   (s/required-key :owner)      IdType
+   (s/required-key :version)    s/Int
+   (s/optional-key :properties) [ModelProperty]})
+
+(def DataItem
+  "A data item"
+  {(s/required-key :id)        IdType
+   (s/required-key :name)      s/Str
+   (s/required-key :publisher) IdType ;; a user or org
+   (s/optional-key :version)   s/Int})
+
+(def DataItemEntry
+  "Used to isolate a data item ID"
+  {(s/required-key :data-item) DataItem})
+
+(def ModelInputCategory
+  "Inputs into the model"
+  {(s/required-key :id)   IdType
+   (s/required-key :name) s/Str})
+
+(def ModelOutputCategory
+  "Inputs into the model"
+  {(s/required-key :id)   IdType
+   (s/required-key :name) s/Str})
+
+(def ModelInput
+  "An input category with a data item"
+  {ModelInputCategory (s/maybe DataItemEntry)})
+
+(def ModelOutput
+  "An output category with a data item"
+  {ModelOutputCategory (s/maybe (s/cond-pre [DataItemEntry] DataItemEntry))})
+
+(def ModelInfo
+  "More in-depth information about a Model"
+  {(s/required-key :model)    Model
+   (s/required-key :inputs)  [ModelInput]
+   (s/required-key :outputs) [ModelOutput]})
+
+(def Tag
+  "A tag is an annotated symlink to a forecast id and version"
+  {(s/required-key :id)   IdType
+   (s/required-key :name) s/Str
+   (s/optional-key :description) s/Str
+   (s/required-key :forecast-id) IdType
+   (s/required-key :forecast-version) s/Int})
+
+(def Forecast
+  "Forecast"
+  {(s/required-key :series)        IdType
+   (s/required-key :name)          s/Str
+   (s/required-key :owner)         IdType
+   (s/required-key :version)       s/Int
```

Comment:
would we need a separate structure ForecastVersion?
when we do a new version of forecast, we post the metadata of that forecast (probably in terms of input data sets in this version) linked to a ForecastId (just thinking aloud here)